src/features/options/effects/EffectPreselectedOptionIndex.tsx

```diff
@@ -30,12 +32,17 @@ export function EffectPreselectedOptionIndex({ preselectedOption, valueType, opt
   const shouldSelectOptionAutomatically =
     !hasValue && !hasSelectedInitial.current && preselectedOption !== undefined && isNodeHidden !== true;
 
-  NodesInternal.useEffectWhenReady(() => {
+  useEffect(() => {
     if (shouldSelectOptionAutomatically) {
+      if (nodeStore.getState().readiness !== NodesReadiness.Ready) {
+        requestAnimationFrame(() => setForceUpdate((prev) => prev + 1));
+        return;
+      }
```

Comment:
That seems like a quite the machine gun of re-renders 🤯  The original `useEffectWhenReady` was a bit complex maybe, what was the issue here?

Reply:
When cy.clock is called in testPdf? Should any of these effects be running in pdf mode?

Reply:
They didn't get a chance to run before `?pdf=1` got added, so yes, they do. We don't have any checks to make sure we don't try to write to a datamodel when in PDF mode. For a while now the PDF generation would just end up crashing instead. I don't think it's better to introduce those checks either, because we'd just be masking configuration mistakes in the apps - and we'd have to remember to add that check as early as possible everywhere the data model can be changed (which is pretty much all over the place).

Reply:
I reverted these changes, ran the pdf test (which I believe is the one that failed) in a loop. After 10+ iterations it hasn't failed, so I would think it's all OK now. :raised_hands: